lib/rules/custom-property-pseudo-selectors/custom-property-pseudo-selectors.js

```diff
@@ -0,0 +1,67 @@
+const stylelint = require('stylelint');
+const valueParser = require('postcss-value-parser');
+
+const ruleName = 'terra/custom-property-pseudo-selectors';
+
+const messages = stylelint.utils.ruleMessages(ruleName, {
+  expected(property, selectors) {
+    return `Expected ${property} to include all ascestor psuedo selectors in order: [${selectors}]`;
+  },
+});
+
+/**
+ * Recursively traverses the css tree and returns the full selector of a node.
+ * @param {PostCSS Node} node - The post css node.
+ * @param {string} [path=''] - The accumulated selector.
+ * @return {string} - The full selector path to the node.
+ */
+const fullSelector = (node, path = '') => {
+  const { parent, selector } = node;
+
+  if (!node.parent || !selector || node.type === 'root') {
+    return path;
+  }
+
+  return fullSelector(parent, `${selector} ${path}`);
+};
+
+/**
+ * Requires custom properties defined within var functions to contain all pseudo selectors in order.
+ * @param {boolean} enabled - Indicates whether or not the rule is enabled.
+ * @return {func} - A rule function.
+ */
+const rule = enabled => (root, result) => {
+  if (!enabled) {
+    return;
+  }
+
+  root.walkDecls((decl) => {
+    const { parent } = decl;
+
+    // Matches all selectors in the ascestor tree and converts them into an ordered regex.
+    const pseudo = fullSelector(parent).match(/:(active|before|after|checked|disabled|empty|focus|hover)/g) || [];
```

Comment:
Should we include `visited`? we have [color themeable](https://github.com/cerner/terra-core/blob/master/packages/terra-hyperlink/src/Hyperlink.module.scss#L13) on a `:visited` hyperlink

Reply:
Added: https://github.com/cerner/stylelint-config-terra/commit/613bbc3a0da0ac2f12ef320c55ad3b82250b7305